src/Microsoft.Health.Fhir.Liquid.Converter/Filters/DateFilters.cs

```diff
@@ -15,114 +15,78 @@ namespace Microsoft.Health.Fhir.Liquid.Converter
-                var timeZoneMinute = int.Parse(groups["timeZoneMinute"].Value);
-                result += groups["sign"].Value + new TimeSpan(timeZoneHour, timeZoneMinute, 0).ToString(@"hh\:mm");
+                dateTimeObject = new PartialDateTime(input, DateTimeType.Hl7v2);
             }
-
-            if (!DateTime.TryParse(result, out var parsedResult))
+            catch (Exception)
             {
                 throw new RenderException(FhirConverterErrorCode.InvalidDateTimeFormat, string.Format(Resources.InvalidDateTimeFormat, input));
             }
 
-            var dateTimeFormat = parsedResult.Millisecond == 0 ? "yyyy-MM-ddTHH:mm:ss%K" : "yyyy-MM-ddTHH:mm:ss.fff%K";
-            return timeZoneHandling?.ToLower() switch
-            {
-                "preserve" => result,
-                "utc" => TimeZoneInfo.ConvertTimeToUtc(parsedResult).ToString(dateTimeFormat),
-                "local" => parsedResult.ToString(dateTimeFormat),
-                _ => throw new RenderException(FhirConverterErrorCode.InvalidTimeZoneHandling, Resources.InvalidTimeZoneHandling),
-            };
-        }
-
-        public static string Now(string input, string format = "yyyy-MM-ddTHH:mm:ss.FFFZ")
-        {
-            return DateTime.UtcNow.ToString(format);
+            dateTimeObject.ConvertToDate();
+            return dateTimeObject.ToFhirString(timeZoneHandling);
         }
 
-        private static string ConvertDate(string input, GroupCollection groups)
+        public static string FormatAsDateTime(string input, string timeZoneHandling = "local")
         {
-            var year = groups["year"].Success ? int.Parse(groups["year"].Value) : 1;
-            var month = groups["month"].Success ? int.Parse(groups["month"].Value) : 1;
-            var day = groups["day"].Success ? int.Parse(groups["day"].Value) : 1;
-
-            try
+            if (string.IsNullOrEmpty(input))
             {
-                var dateTime = new DateTime(year, month, day);
-                return groups["year"].Success switch
-                {
-                    true when groups["month"].Success && groups["day"].Success => dateTime.ToString("yyyy-MM-dd"),
-                    true when groups["month"].Success => dateTime.ToString("yyyy-MM"),
-                    true => dateTime.ToString("yyyy"),
-                    _ => throw new RenderException(FhirConverterErrorCode.InvalidDateTimeFormat, string.Format(Resources.InvalidDateTimeFormat, input))
-                };
+                return input;
             }
-            catch (Exception)
+
+            if (!TimezoneHandlingMethods.Contains(timeZoneHandling))
             {
-                throw new RenderException(FhirConverterErrorCode.InvalidDateTimeFormat, string.Format(Resources.InvalidDateTimeFormat, input));
+                throw new RenderException(FhirConverterErrorCode.InvalidTimeZoneHandling, Resources.InvalidTimeZoneHandling);
```

Comment:
Do we need to check this for `AddHyphensDate` as well?

Reply:
Thanks, added here.